x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/rest/action/RestSamlInitiateSingleSignOnAction.java

```diff
@@ -65,6 +65,7 @@ public RestResponse buildResponse(SamlInitiateSingleSignOnResponse response, XCo
                         builder.startObject();
                         builder.field("post_url", response.getPostUrl());
                         builder.field("saml_response", response.getSamlResponse());
+                        builder.field("error", response.getError());
```

Comment:
Is the contract that if `error` is `null` then it's a success and if it's non-null then there's a failure?

Reply:
yes. I contemplated adding a boolean `isSuccess` field but felt like this extra check wouldn't make much sense on the consumer side but I don't feel strongly about it if you think it makes this clearer/easier to consume.

Reply:
It feels a little weird, but I can't think of anything better, unless we add the saml status as well.

Reply:
I ended up doing just that after all